airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/documentation/documentation.py

```diff
@@ -0,0 +1,605 @@
+
+        actual_contents = documentation.section(self.PREREQUISITES)
+        if len(actual_contents) > 1:
+            return [f"Documentation has more than one {self.PREREQUISITES} section. Please check it."]
+
+        missing_fields: List[str] = []
+        section_content = documentation.section(self.PREREQUISITES)[0].lower()
+
+        spec = actual_connector_spec.get("connectionSpecification") or actual_connector_spec.get("connection_specification")
+        required_titles, has_credentials = required_titles_from_spec(spec)  # type: ignore
+
+        for title in required_titles:
+            if title.lower() not in section_content:
+                missing_fields.append(title)
+
+        if has_credentials:
+            # credentials has specific check for keywords as we have a lot of ways how to describe this step
+            credentials_validation = [k in section_content for k in self.CREDENTIALS_KEYWORDS]
+            if True not in credentials_validation:
+                missing_fields.append(f"credentials")
+
+        return missing_fields
+
+    def _run(self, connector: Connector) -> CheckResult:
+        if not connector.documentation_file_path or not connector.documentation_file_path.exists():
+            return self.fail(
+                connector=connector,
+                message="Could not check documentation structure as the documentation file is missing.",
+            )
+
+        if not connector.documentation_file_path.read_text():
+            return self.fail(
+                connector=connector,
+                message="Documentation file is empty",
+            )
+
+        # check_prerequisites uses spec content from file, not from spec command,
+        # which possible can lead to incorrect testing, for now it works for connectors with sl>=300.
+        # But if someone faced with unexpected behavior of this test it's better to disable it.
+
+        errors = self.check_prerequisites(connector)
+
+        if errors:
+            return self.fail(
+                connector=connector,
+                message=f"Missing descriptions for required spec fields: {'. '.join(errors)}",
+            )
+        return self.pass_(
+            connector=connector,
+            message="All required fields from spec are present in the connector documentation.",
+        )
+
+
+class CheckSection(CheckDocumentationContent):
+    required = True
+    expected_section_index = 0
+
+    @property
+    def name(self) -> str:
+        return f"'{self.header}' section of the documentation follows our guidelines"
+
+    @property
+    def description(self) -> str:
+        templates = TemplateContent("<CONNECTOR_NAME_FROM_METADATA>").section(self.header)
+        if len(templates) > 1:
+            template = templates[1]
+        else:
+            template = templates[0]
+
+        return generate_description("section_content_description.md", {"header": self.header, "template": template})
```

Comment:
Again, I don't get why we need a template here. Can't we use a static file describing the expected sections?

Reply:
I just didn't want to duplicate section description here and in template file. For me it's easier when it's generated from template instead of copy pasting as string to description.